### PR TITLE
27 article list

### LINKED ISSUE #27: Create WofArticleList component
Component will be used for presenting WIki articles as list in search, list or user pages.
Design based on Figma (if #153 is closed )

### PR DESCRIPTION
**Pull request by karolkozlo**
closes #27

### CODE REVIEW

**Comment by karolkozlo** [comment]
I tested it with this code in App.vue

```
<template>
  <main>
    <wof-header/>
    <div class="main-wrapper">
      <router-view msg="This is a basic frontend">
          <wof-article-list :articles="articles" :isButtonVisible="true" @add="addArticle"></wof-article-list>
      </router-view>
    </div>
    <wof-footer/>
  </main>
</template>

<script>
import WofFooter from './components/WofFooter.vue';
import WofHeader from './components/WofHeader.vue';
import WofArticleList from './components/WofArticleList.vue';

export default {
  components: { WofFooter, WofHeader, WofArticleList },
  name: "App",
  data() {
    return {
      articles: [
        {
          id: 1,
          title: 'Some interesting article1',
          linkToContents: 'https://en.wikipedia.org/wiki/Potato',
          dateAdded: '24.12.2021',
          lastModification: '27.12.2021',
          numberOfViews: 3401,
          numberOfEditions: 7
        },
        {
          id: 2,
          title: 'Best City at Mediterranean Sea Ever!!!!1!',
          linkToContents: 'https://en.wikipedia.org/wiki/Constantinople',
          dateAdded: '25.12.2021',
          lastModification: '28.12.2021',
          numberOfViews: 6543,
          numberOfEditions: 10
        },
        {
          id: 3,
          title: 'I know it from college',
          linkToContents: 'https://en.wikipedia.org/wiki/Turing_machine',
          dateAdded: '27.12.2021',
          lastModification: '22.12.2021',
          numberOfViews: 245,
          numberOfEditions: 1
        },
      ]
    }
  },
  methods: {
    addArticle(articleID) {
      console.log(articleID);
    }
  }
};
</script>

File: frontend/src/components/WofArticleList.vue [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                   Date added
+                </th>
+                <th class="wof-article-list__header">
+                    Last Edition
+                </th>
+                <th class="wof-article-list__header wof-article-list__header--small">
+                    Number of viewers (this week)
+                </th>
+                <th class="wof-article-list__header wof-article-list__header--small">
+                    Number of editions (this month)
```
Comment (kpagacz) [inline frontend/src/components/WofArticleList.vue:18]: Can we shorten this to just: views (last week) and edits (last month)?

File: frontend/src/components/WofArticleList.vue [hunk 2]
```diff
@@ -0,0 +1,133 @@
+<template>
```
Comment (kpagacz) [inline frontend/src/components/WofArticleList.vue:1]: This file looks completely unformatted. I have run vetur on this and got a bunch of changesi in indentation, and spaces between if/for and brackets. Can you make sure a formatter is run on this?

**Review by kpagacz — CHANGES_REQUESTED** [review]
Some styling comments mainly. It does what it says on the tin!
image

File: frontend/src/components/WofArticleList.vue [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+</template>
+
+<script>
+export default {
+    name: 'WofArticleList',
+    props: {
+        articles: {
+            type: Array,
+            validator: (list) => {
+                for(let i=0; i<list.lenght; i++) {
```
Comment (robert10290) [inline frontend/src/components/WofArticleList.vue:52]: typo here

**Review by kpagacz — APPROVED** [review]
LGTM
